presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -0,0 +1,86 @@
+package me.androidbox.presentation.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import me.androidbox.presentation.login.screen.LoginScreen
+import me.androidbox.presentation.login.screen.RegisterScreen
+import me.androidbox.presentation.login.viewmodel.LoginViewModel
+import me.androidbox.presentation.login.viewmodel.RegisterViewModel
+
+@Composable
+fun NavigationGraph(
+    navHostController: NavHostController
+) {
+    NavHost(
+        navController = navHostController,
+        startDestination = Screen.LoginScreen.route
+    ) {
+
+        /* LoginScreen */
+        composable(
+            route = Screen.LoginScreen.route
+        ) {
+            val loginViewModel: LoginViewModel = hiltViewModel()
+            val loginResponseState = loginViewModel.loginResponseState.collectAsState()
+
+            LoginScreen(
+                loginResponseState = loginResponseState,
+                loginScreenEvent = { loginEvent ->
+                    loginViewModel.onLoginEvent(loginEvent)
+                },
+                onSignUpClicked = {
+                    /* Signup clicked, navigate to register screen */
+                    navHostController.navigate(route = Screen.RegisterScreen.route)
+                },
+                onLoginSuccess = { login ->
+                    loginViewModel.saveCurrentUserDetails(login)
+                    /* TODO Navigate to the agenda screen (not implemented yet) */
+                },
+                loginScreenState = loginViewModel.loginScreenState.collectAsState().value
+            )
+        }
```

Comment:
For `onSignUpClicked` and `onLoginSuccess` I think I should keep them here as this is where I want to do the navigation to other screens.